pkg/exec/helpers.go

```diff
@@ -110,10 +110,11 @@ func RunWithStdoutReader(cmd Cmd, readerFunc func(io.Reader) error) error {
 	}()
 
 	err = cmd.Run()
+	pw.Close()
```

Comment:
this is smart, if we close it here we can't write thus we can't break the pipe, right?

Reply:
Er.
So this is an os.Pipe, which would make more sense if we didn't have the multiwriter thing underneath etc...

Anyhow, it should get closed when the process exits if it were a real pipe, go would have passed it directly as an fd and closed this process's duplicate immediately after starting the process, retaining only the end this process uses.

We should close it here to ensure the reader can't hang on a still open pipe waiting for more.